sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java

```diff
@@ -102,43 +116,35 @@ public JacksonAdapter() {
      * are pre-configured for Azure serialization needs, but only outer mapper capable of flattening and populating
      * additionalProperties. Outer mapper is used by {@code JacksonAdapter} for all serialization needs.
      * <p>
-     * Register modules on the outer instance to add custom (de)serializers similar to {@code new JacksonAdapter((outer,
-     * inner) -> outer.registerModule(new MyModule()))}
+     * Register modules on the outer instance to add custom (de)serializers similar to
+     * {@code new JacksonAdapter((outer, inner) -> outer.registerModule(new MyModule()))}
      *
      * Use inner mapper for chaining serialization logic in your (de)serializers.
      *
      * @param configureSerialization Applies additional configuration to outer mapper using inner mapper for module
      * chaining.
+     * @throws UnsupportedOperationException Always, as this is no longer supported.
+     * @deprecated Use {@link #createDefaultSerializerAdapter()} instead as modified
+     * {@link JacksonAdapter JacksonAdapters} are no longer supported.
      */
+    @Deprecated
     public JacksonAdapter(BiConsumer<ObjectMapper, ObjectMapper> configureSerialization) {
-        Objects.requireNonNull(configureSerialization, "'configureSerialization' cannot be null.");
-        this.headerMapper = ObjectMapperShim.createHeaderMapper();
-        this.mapper = ObjectMapperShim.createJsonMapper(ObjectMapperShim.createSimpleMapper(),
-            (outerMapper, innerMapper) -> captureRawMappersAndConfigure(outerMapper, innerMapper, configureSerialization));
-    }
-
-    /**
-     * Temporary way to capture raw ObjectMapper instances, allows to support deprecated simpleMapper() and
-     * serializer()
-     */
-    private void captureRawMappersAndConfigure(ObjectMapper outerMapper, ObjectMapper innerMapper,
-        BiConsumer<ObjectMapper, ObjectMapper> configure) {
-        this.rawOuterMapper = outerMapper;
-        this.rawInnerMapper = innerMapper;
-
-        configure.accept(outerMapper, innerMapper);
+        throw LOGGER.logExceptionAsError(new UnsupportedOperationException("new "
```

Comment:
This will be reverted, used to ensure all usages were removed. Same for all other deprecated APIs in this class.